antarest/launcher/service.py

```diff
@@ -603,74 +601,28 @@ def download_output(self, job_id: str) -> FileDownloadTaskDTO:
-                    config_path=Path(),
-                    username=slurm_config.username,
-                    hostname=slurm_config.hostname,
-                    port=slurm_config.port,
-                    private_key_file=slurm_config.private_key_file,
-                    key_password=slurm_config.key_password,
-                    password=slurm_config.password,
-                )
-                partition = slurm_config.partition
-                allocated_cpus, cluster_load, queued_jobs = calculates_slurm_load(ssh_config, partition)
-                args = {
-                    "allocatedCpuRate": allocated_cpus,
-                    "clusterLoadRate": cluster_load,
-                    "nbQueuedJobs": queued_jobs,
-                    "launcherStatus": "SUCCESS",
-                }
-                return LauncherLoadDTO(**args)
-            else:
-                raise KeyError("Default launcher is slurm but it is not registered in the config file")
-
-        # local load calculation
-        local_used_cpus = sum(
-            LauncherParametersDTO.from_launcher_params(job.launcher_params).nb_cpu or 1
-            for job in self.job_result_repository.get_running()
-        )
 
-        # The cluster load is approximated by the percentage of used CPUs.
-        cluster_load_approx = min(100.0, 100 * local_used_cpus / (os.cpu_count() or 1))
+        return self.config.launcher.get_load(self.job_result_repository.get_running(), launcher_id)
 
-        args = {
-            "allocatedCpuRate": cluster_load_approx,
-            "clusterLoadRate": cluster_load_approx,
-            "nbQueuedJobs": 0,
-            "launcherStatus": "SUCCESS",
-        }
-        return LauncherLoadDTO(**args)
-
-    def get_solver_versions(self, solver: str) -> List[str]:
+    def get_solver_versions(self, launcher_id: Optional[str]) -> List[str]:
         """
         Fetch the list of solver versions from the configuration.
 
         Args:
-            solver: name of the configuration to read: "default", "slurm" or "local".
+            launcher_id: id of the configuration to read.
 
         Returns:
             The list of solver versions.
             This list is empty if the configuration is not available.
 
         Raises:
-            KeyError: if the configuration is not "default", "slurm" or "local".
+            InvalidConfigurationError: if the launcher doesn't exist in the configuration.
         """
-        local_config = self.config.launcher.local
-        slurm_config = self.config.launcher.slurm
-        default_config = self.config.launcher.default
-        versions_map = {
-            "local": sorted(local_config.binaries) if local_config else [],
-            "slurm": sorted(slurm_config.antares_versions_on_remote_server) if slurm_config else [],
-        }
-        versions_map["default"] = versions_map[default_config]
-        return versions_map[solver]
+        return self.config.launcher.get_solver_versions(launcher_id)
```

Comment:
In relation with previous comment: here we should ask the the launcher implementation instead:
```python
return self.launchers[cluster_id].get_solver_versions()
```